docs/reuse/reference/extensions/integrations.rst

```diff
@@ -22,206 +22,248 @@ integrations, for internal purposes.
+To add one of these relations, e.g., PostgreSQL, in the
+project file, include the appropriate ``requires`` block and
 integrate with |juju_integrate_postgresql| as usual.
 
-After the integration has been established, the connection string will
-be available as an environment variable. Integration with PostgreSQL,
-MySQL, MongoDB or Redis provides the string as the
-``POSTGRESQL_DB_CONNECT_STRING``, ``MYSQL_DB_CONNECT_STRING``,
-``MONGODB_DB_CONNECT_STRING`` or ``REDIS_DB_CONNECT_STRING`` environment
-variables respectively. Furthermore, the following environment variables
-will be provided to your |framework| application for integrations with
-PostgreSQL, MySQL, MongoDB or Redis:
-
-- ``<integration>_DB_SCHEME``
-- ``<integration>_DB_NETLOC``
-- ``<integration>_DB_PATH``
-- ``<integration>_DB_PARAMS``
-- ``<integration>_DB_QUERY``
-- ``<integration>_DB_FRAGMENT``
-- ``<integration>_DB_USERNAME``
-- ``<integration>_DB_PASSWORD``
-- ``<integration>_DB_HOSTNAME``
-- ``<integration>_DB_PORT``
-- ``<integration>_DB_NAME``
-
-Here, ``<integration>`` is replaced by ``POSTGRESQL``, ``MYSQL``
-``MONGODB`` or ``REDIS`` for the relevant integration.
-
-The provided SAML environment variables are as follows:
-
-- ``SAML_ENTITY_ID`` (required)
-- ``SAML_METADATA_URL`` (required)
-- ``SAML_SINGLE_SIGN_ON_REDIRECT_URL`` (required)
-- ``SAML_SIGNING_CERTIFICATE`` (required)
-
-The S3 integration creates the following environment variables that you
-may use to configure your |framework| application:
-
-- ``S3_ACCESS_KEY`` (required)
-- ``S3_SECRET_KEY`` (required)
-- ``S3_BUCKET`` (required)
-- ``S3_REGION``
-- ``S3_STORAGE_CLASS``
-- ``S3_ENDPOINT``
-- ``S3_PATH``
-- ``S3_API_VERSION``
-- ``S3_URI_STYLE``
-- ``S3_ADDRESSING_STYLE``
-- ``S3_ATTRIBUTES``
-- ``S3_TLS_CA_CHAIN``
-
-The RabbitMQ integration creates the connection string in the
-environment variable ``RABBITMQ_CONNECT_STRING``. Furthermore, the
-following environment variables may be provided, derived from the
-connection string:
-
-- ``RABBITMQ_SCHEME``
-- ``RABBITMQ_NETLOC``
-- ``RABBITMQ_PATH``
-- ``RABBITMQ_PARAMS``
-- ``RABBITMQ_QUERY``
-- ``RABBITMQ_FRAGMENT``
-- ``RABBITMQ_USERNAME``
-- ``RABBITMQ_PASSWORD``
-- ``RABBITMQ_HOSTNAME``
-- ``RABBITMQ_PORT``
-- ``RABBITMQ_VHOST``
-
-The Tracing integration creates the following environment variables
-that you can use to configure your application:
-
-- ``OTEL_EXPORTER_OTLP_ENDPOINT``
-- ``OTEL_SERVICE_NAME``
-
-The provided SMTP environment variables are as follows:
-
-- ``SMTP_HOST``
-- ``SMTP_PORT``
-- ``SMTP_USER``
-- ``SMTP_PASSWORD_ID``
-- ``SMTP_AUTH_TYPE``
-- ``SMTP_TRANSPORT_SECURITY``
-- ``SMTP_DOMAIN``
-
-The provided OpenFGA environment variables are as follows:
-
-- ``FGA_STORE_ID``
-- ``FGA_TOKEN``
-- ``FGA_GRPC_API_URL``
-- ``FGA_HTTP_API_URL``
+Environment variables
+~~~~~~~~~~~~~~~~~~~~~
+
+After the relation has been established, environment variables will be
+provided to your |framework| app:
```

Comment:
```suggestion
Each relation adds its own environment variables to your |framework| app. Some
must be set for the relation to function.
```

We're combining both remote past and future tense in the same sentence, which is a peeve of mine. I also added a hint about what "required" means.